signatures.md

```diff
@@ -52,163 +49,359 @@ Note: `Message` is used here as the object being signed, but this interface shou
+Currently, Filecoin uses ECDSA (curve secp256k1) or BLS (curve BLS12-381) signatures to fulfill the above interface. All signatures on messages, blocks, and tickets use the same scheme and format.
 
-- Point compression Go - <https://github.com/btcsuite/btcd/blob/86fed781132ac890ee03e906e4ecd5d6fa180c64/btcec/signature.go#L338>
+### Wire Format
 
-- Create recoverable sig - <https://github.com/ipsn/go-secp256k1/blob/9d62b9f0bc52d16160f79bfb84b2bbf0f6276b03/libsecp256k1/include/secp256k1_recovery.h#L70>
+What bits are on the wire and in what order/format. We are currently adhering to libsecp256k1 serialization for ECDSA but will not list that wire format here. We do point to the code where an authoritative format is defined.
 
-- Recid - <https://github.com/bitcoin-core/secp256k1/blob/314a61d72474aa29ff4afba8472553ad91d88e9d/src/ecdsa_impl.h#L288>
+**ECDSA**
 
-- Serialize - <https://github.com/bitcoin-core/secp256k1/blob/314a61d72474aa29ff4afba8472553ad91d88e9d/src/ecdsa_impl.h#L177>
+Follow: [secp256k1 Reference](https://github.com/bitcoin-core/secp256k1/blob/314a61d72474aa29ff4afba8472553ad91d88e9d/src/ecdsa_impl.h#L177)
 
-- Should elliptic curve points always be compressed in Marshal?
+**BLS**
 
-- Recover is math trick trading CPU for space to generate sk*G
+Follow: [IETF BLS Signature Scheme - Type Conversions](https://tools.ietf.org/html/draft-boneh-bls-signature-00#section-2.6.2)
 
-- Serialization should account for :
+## Github References
 
-- - Should there be point compression? 
+<https://github.com/filecoin-project/specs/issues/131>
 
-  - Point compression (x,y); len(x) = n bits → x, recovery_bit; len(x||r_bit) = n+1 bits
+[go-filecoin SignedMessage](https://github.com/filecoin-project/go-filecoin/blob/master/types/signed_message.go)
 
-  - - Easy to check add; use (mod2) or check last bit
+[go-filecoin signer](https://github.com/filecoin-project/go-filecoin/blob/master/types/signer.go)
 
-  - Sig = (r,s) - s is standard but r could be (rx,ry) or rx or r all of these can be used to generate r*G
+## References & Notes
 
-  - Recovery() can be used for (1) finding r*G from (r,s) and (2) pk = sk*G from SignatureBytes
+Signatures are not sent over the wire on their own, instead they are encapsulated in in a [SignedMessage](https://github.com/filecoin-project/go-filecoin/blob/master/types/signed_message.go#L22-L28). Each `SignesMessage` contains a `Message`, `Address`, and a signature. `SignedMessage` is sent over the wire by marshaling it to cbor and send, once it is received it is unmarshaled form [cbor](http://cbor.io/) back into the `SignedMessage` Structure.
+[Code for marshaling and unmarshaling SignedMessages](https://github.com/filecoin-project/go-filecoin/blob/master/types/signed_message.go#L22-L38)
```

Comment:
let's drop any references to go-filecoin code, in favor of pointing to things in the spec.

Reply:
removed